### ISSUE TITLE
Importance does not take a parent's due date into account

### ISSUE THREAD

**Issue by jancp** [bram85/topydo#181]
Hello!

I'm not sure if this is a missing feature or a bug, but child tasks do not seem to inherit their parent's due date. Consider this to-do list:

```
(B) 2017-05-31 Respond to Alice' question regarding the cucumber salad recipe due:2017-06-07 id:1
(B) 2017-05-31 Confirm with Bob that cucumber salad recipe is correct p:1
```
We first want to check with Bob before we answer to Alice, so we make the Bob task a child of the Alice task. So far, so good.

However, if we now add another task there's a little problem:

```
(B) 2017-05-31 Make a bowl of cucumber salad for the garden party due:2017-06-14
```
Now, the new task is in front of the task with Bob.

I think this is wrong. If a child has no due date then topydo should use the parent's due date instead.

Cheers!

Jan

**Comment by bram85** [comment]
Instead of sorting by normal importance, there's also the [average importance](https://topydo.org/#AverageImportance) that takes the importance of the parents into account. That would raise the importance of the child task a bit.

1.5 years ago I started a branch that would [implement](https://github.com/bram85/topydo/compare/due-parent) the behavior you described, but in the end I didn't merge it because I didn't like the behavior so much. I tend to keep the usage of due dates to a minimum, and there was no need to duplicate due dates among children because the parent holds the 'true' due date (single point of definition). The average importance would make sure that the child item pops up whenever it becomes relevant, regardless presence/value of its due date.

Nevertheless, I implemented it as an option. So I could try to re-apply the patch again on the current codebase and leave its behavior disabled by default.

**Comment by jancp** [comment]
> Instead of sorting by normal importance, there's also the average importance that takes the importance of the parents into account. That would raise the importance of the child task a bit.

Indeed, it does. Even though it's only a bit (as long as the due date is still in the future).

> So I could try to re-apply the patch again on the current codebase and leave its behavior disabled by default.

I would really appreciate that. And I hope others would too. :)

As a bounty I could give you the recipe for the cucumber salad. ;)

Cheers!

Jan
